### PR TITLE
feat(sdk): Add task workspace support

### LINKED ISSUE #1150: add workspace config at task level
/kind feature

**Description:**
[A clear and concise description of what your proposal. What problem does it solve?]

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #1150 

**Description of your changes:**

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp_tekton/compiler/compiler.py [hunk 1]
```diff
@@ -1039,6 +1040,12 @@ def map_cel_vars(a):
             ],
             'taskSpec': template['spec'],
           }
+        if template['spec'].get("workspaces"):
+          workspaces_spec = []
+          for item in template['spec']['workspaces']:
+            workspaces_spec.append({"name": item["name"], "workspace": item["name"]})
```
Comment (yhwang) [inline sdk/python/kfp_tekton/compiler/compiler.py:1046]: workspace's value?

Comment (Tomcli) [reply sdk/python/kfp_tekton/compiler/compiler.py:1046]: template['spec']['workspaces'] is a list of dict like below, on the pipelinetask level we just need to have the name of the workspace and which workspace we want to refer to in the pipeline definition. For simplicity, I expect all the user defined workspaces will have the same name within the same pipeline.

```
taskSpec:
  workspaces:
  - name: new-ws
    readOnly: true
```

File: sdk/python/tests/compiler/testdata/tekton_pipeline_conf.yaml [hunk 2]
```diff
@@ -48,12 +48,22 @@ spec:
...
           - -c
           image: busybox
+        workspaces:
+        - name: new-ws
+          readOnly: true
         metadata:
           labels:
             pipelines.kubeflow.org/cache_enabled: "true"
           annotations:
+            workspaces: '{"new-ws": {"readOnly": true}}'
```
Comment (yhwang) [inline sdk/python/tests/compiler/testdata/tekton_pipeline_conf.yaml:58]: do we need this annotation in the output yaml?

Comment (Tomcli) [reply sdk/python/tests/compiler/testdata/tekton_pipeline_conf.yaml]: thanks, I updated the code to remove this annotation.

**Review by yhwang — APPROVED** [review]
/lgtm

**Pull request merged**